src/rm_topk.c

```diff
@@ -198,7 +200,12 @@ void TopKRdbSave(RedisModuleIO *io, void *obj) {
     RedisModule_SaveStringBuffer(io, (const char *)topk->heap,
                                  topk->k * sizeof(HeapBucket));
     for(uint32_t i = 0; i < topk->k; ++i) {
-        RedisModule_SaveStringBuffer(io, topk->heap[i].item, strlen(topk->heap[i].item) + 1);
+        if(topk->heap[i].item != NULL) {
+            RedisModule_SaveStringBuffer(io, topk->heap[i].item, strlen(topk->heap[i].item) + 1);
+        } else {
+            char *emptystr = "\n";
+            RedisModule_SaveStringBuffer(io, emptystr, 1);
```

Comment:
Serialize an empty string like:
`RedisModule_SaveStringBuffer(io, "", 1);`
Which causes the string to consist of a single NULL byte. The method here has 2 bytes, so serialization still turns a bit odd:

```
$ redis-cli TOPK.RESERVE test 3 5 4 0.9
OK
$ redis-cli TOPK.LIST test
1) (nil)
2) (nil)
3) (nil)
$ redis-cli debug reload
OK
$ redis-cli TOPK.LIST test
1)

2)

3)
#
```

Reply:
Solved